docs/current_state_specification.md

```diff
@@ -0,0 +1,153 @@
+| name | string | Name of a hardware node |
+| parameters | JSON array | List of key/value parameters for a hardware node |
+| addressable | boolean | Enables hardware node addressability |
+| **Software node properties** | | |
+| id | string | Identifier of the software node |
+| version | string | Version of the software node |
+| name | string | Name of the software node |
+| parameters | JSON array | List of key/value parameters for a software node |
+| type | string | Type of the software node. The supported types are listed below |
+| **Parameter properties** | | |
+| key | string | Key of the parameter |
+| value | string | Value of the parameter |
+| **Association parameter** | | |
+| sourceId | string | Identifier of the source node of the association |
+| targetId | string | Identifier of the target node of the association |
+
+### Supported software types
+
+The list of the supported software types :
+
+| Type | Description |
+| - | - |
+| IMAGE | Represents an image software type |
+| RAW | Represents a raw bytes software type |
+| DATA | Represents a data software type |
+| APPLICATION | Represents an application software type |
+| CONTAINER | Represents a container software type |
+
+### Device Inventory Graph Representation
+
+The diagram below represents the Device Inventory graph and the links between the software and hardware nodes.
+
+The software nodes are organized in the graph at different levels. At the root level stands the main Update Manager software node, which represents the Update Manager component. At the second level are placed the root nodes for each supported domain agent, which are linked with associations to the Update Manager node. At the last level in the hierarchy are placed the domain-specific software nodes, representing the domain components. These software nodes can be modeled the in a tree-based structure if the internal domain-specific representation is more complex.
+
+The hardware nodes do not follow any strict hierarchy and can be linked to any hardware or software node. Any cycles between the nodes are not allowed and prevented.
+
+![Device inventory](/_assets/device-inventory.svg)
+
+### Current State Data Model Example
+
+The following data structure is a holistic example view of a device current state:
+```json
+{
+	"hardwareNodes": [
+		{
+			"id": "cOffee",
+			"version": "rev2",
+			"name": "OWASYS box",
+			"parameters": [
+				{
+					"key": "cpu-arch",
+					"value": "armv7"
+				}
+			]
+		}
+	],
+	"softwareNodes": [
+		{
+			"id": "update-manager",
+			"version": "1.0.0",
+			"name": "Update Manager",
+			"type": "APPLICATION"
+		},
+		{
+			"id": "containers-update-agent",
+			"version": "1.0",
+			"name": "Containers Update Agent",
+			"type": "APPLICATION",
+			"parameters": [
+				{
+					"key": "domain",
+					"value": "containers"
+				},
+				{
+					"key": "container_registry",
+					"value": "ghcr.io"
+				}
+			]
+		},
+		{
+			"id": "containers:helloworld",

```

Comment:
My assumption would be that the full container name is then `[key:container_registry]:[id]` so in this example `ghcr.io/conainers:helloworld1.0.0" . Did I get that right? How would you then deal with container coming from different container registries?
If the in this example the `helloworld` is already the tag where is the version field then used?

Reply:
You are right. The given example does not match the actual implementation inside the containers update agent of Kanto container management component. It was meant to be a simplified example, just to illustrate the associations between update manager node and update agent nodes, and also update agent nodes and domain-specific nodes (containers, images...). 
Detailed key-value descriptions and examples for _containers_ domain are planned to be published in a separate documentation page - https://github.com/eclipse-kanto/kanto/issues/243.
But maybe it will be good to update the simplified example too, so that it includes at least the mandatory _image_ key-value parameter.

Reply:
Example update - diagram and sample JSON.